scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,688 @@
+import * as React from 'react';
+// @ts-ignore
+import ReactCrop from 'react-image-crop';
+import 'react-image-crop/dist/ReactCrop.css';
+import {ISuperdesk, IArticle} from 'superdesk-api';
+import {isEmpty, omit, pick, isEqual, cloneDeep, flatten} from 'lodash';
+
+import {VideoEditorTools} from './VideoEditorTools';
+import {VideoTimeline} from './VideoTimeline/VideoTimeline';
+import {VideoEditorHeader} from './VideoEditorHeader';
+import {VideoEditorThumbnail} from './VideoEditorThumbnail';
+import {IVideoEditor, IThumbnail, ICrop} from './interfaces';
+
+interface IProps {
+    article: IArticle;
+    superdesk: ISuperdesk;
+    onClose: () => void;
+    onArticleUpdate: (articleUpdate: IArticle) => void;
+}
+
+interface IState {
+    isDirty: boolean;
+    trim: {
+        start: number;
+        end: number;
+    };
+    cropImg: string;
+    thumbnails: Array<IThumbnail>;
+    loading: boolean;
+    loadingText: string;
+    scale: number;
+    videoSrc: string;
+    article: IArticle;
```

Comment:
Because we can't update article via `props.onArticleUpdate`, it seems opening `videoEditor` cause article to lock so it logs error *Article is locked and can't be updated via HTTP directly.The updates will be added to existing diff in article-edit view instead.*, lead to mismatch etag on next api call